iac-common/src/main/java/org/sonar/iac/common/extension/IacSensor.java

```diff
@@ -115,6 +115,13 @@ public void execute(SensorContext sensorContext) {
     statistics.log();
   }
 
+  private List<InputFile> inputFiles(SensorContext sensorContext) {
+    FileSystem fileSystem = sensorContext.fileSystem();
+    FilePredicate predicate = mainFilePredicate(sensorContext);
+    return StreamSupport.stream(fileSystem.inputFiles(predicate).spliterator(), false)
+      .collect(Collectors.toList());
+  }
```

Comment:
That way the predicates are only applied once. Before there were applied in lines 101 and 158.

Reply:
I don't see any duplicate applications before, but looks good nonetheless!